frontend/src/utils/index.js

```diff
@@ -644,25 +644,14 @@ export function omitKeysWithSuffix (obj, suffix) {
   return omit(obj, keys)
 }
 
-export function parseNumericalAbbreviation (abbreviatedNumber) {
-  const regex = /(\d+(?:\.\d+)?)(k|M|B)?/i
-  const match = abbreviatedNumber.match(regex)
-
-  if (!match) {
+export function parseNumberWithMagnitudeSuffix (abbreviatedNumber) {
+  const [, number, suffix] = /^(\d+(?:\.\d*)?)([kmbt]?)$/i.exec(abbreviatedNumber) ?? []
+  if (!number) {
+    logger.error(`Could not parse number with suffix ${abbreviatedNumber} as it does not match regex ^(\\d+(?:\\.\\d*)?)([kmbt]?)$`)
```

Comment:
```suggestion
    logger.error(`Failed to parse ${abbreviatedNumber} because it doesn't follow the required format: a number optionally with a decimal, followed by an optional magnitude suffix ('k', 'm', 'b', 't').`)
```